### PR TITLE
Add the CURRENT_ROLE scalar as an alias to CURRENT_USER

### PR DESCRIPTION
**Pull request by seut**
Improves general SQL Standard compatibility

### CODE REVIEW

File: libs/sql-parser/src/main/antlr/io/crate/sql/parser/antlr/SqlBaseParser.g4 [hunk 1]
```diff
@@ -339,7 +339,8 @@ explicitFunction
     | name=CURRENT_TIMESTAMP
         (OPEN_ROUND_BRACKET precision=integerLiteral CLOSE_ROUND_BRACKET)?           #specialDateTimeFunction
     | CURRENT_SCHEMA                                                                 #currentSchema
-    | (CURRENT_USER | USER)                                                          #currentUser
+    | (CURRENT_USER | CURRENT_ROLE | USER)                                           #currentUser
+    | (CURRENT_USER | CURRENT_ROLE | USER)                                           #currentUser
```
Comment (matriv) [inline libs/sql-parser/src/main/antlr/io/crate/sql/parser/antlr/SqlBaseParser.g4:343]: why is this line added twice?
We can also use `#currentRole` already here, but also find to keep it as is, your call.

**Review by matriv — APPROVED** [review]
LGTM, left a question.

File: libs/sql-parser/src/main/antlr/io/crate/sql/parser/antlr/SqlBaseParser.g4 (hunk 1 above)
Comment (seut) [reply libs/sql-parser/src/main/antlr/io/crate/sql/parser/antlr/SqlBaseParser.g4:343]: Oh, duplicated the line by mistake, thx!

File: libs/sql-parser/src/main/antlr/io/crate/sql/parser/antlr/SqlBaseParser.g4 [hunk 2]
```diff
@@ -864,9 +864,11 @@ nonReserved
     | COPY
     | CURRENT
     | CURRENT_DATE
+    | CURRENT_ROLE
```
[the remark this answers is not in the document]
Comment (seut) [reply libs/sql-parser/src/main/antlr/io/crate/sql/parser/antlr/SqlBaseParser.g4]: Right, I've removed them.

And yes, I think we should follow up here and double check all non-reserved vs. reserved keywords.

**Comment by seut** [comment]
@matriv I've added a breaking change entry related to the new reserved keyword, please check, thx!

File: docs/appendices/release-notes/5.7.0.rst [hunk 3]
```diff
@@ -45,7 +45,8 @@ Version 5.7.0 - Unreleased
 Breaking Changes
 ================
 
-None
+- Added ``CURRENT_ROLE`` as a reserved keyword. If you have a column or table
+  named ``CURRENT_ROLE``, you must escape it using double quotes.
```
Comment (mfussenegger) [inline docs/appendices/release-notes/5.7.0.rst:49]: Could we make this non reserved instead? As long as it doesn't cause ambiguity in the parser there isn't really a downside to it?

Comment (seut) [reply docs/appendices/release-notes/5.7.0.rst:49]: Hm right, will change, thx.

Comment (matriv) [reply docs/appendices/release-notes/5.7.0.rst:49]: Have you seen this: https://github.com/crate/crate/pull/15560#pullrequestreview-1882566899 ?
Is there a benefit of having it non-reserved?
If we do so, we need to fix the logic where we list the reserved keywords (actually how we mark the reserved ones) as the keywords like `CURRENT_ROLE`, `CURRENT_TIME`, etc. are functions and not qualified names, (see my comment above).

Comment (mfussenegger) [reply docs/appendices/release-notes/5.7.0.rst:49]: > Is there a benefit of having it non-reserved?

It wouldn't cause a breaking change.

But that these are functions is a good point. What would happen if they're non-reserved and a user tries to use a `current_role` column name?
